src/utils/exec.c

```diff
@@ -733,13 +734,58 @@ gboolean bd_utils_init_prog_reporting (BDUtilsProgFunc new_prog_func, GError **e
     return TRUE;
 }
 
+/**
+ * bd_utils_init_prog_reporting_thread:
+ * @new_prog_func: (allow-none) (scope notified): progress reporting function to
+ *                                                use on current thread or %NULL
+ *                                                to reset to default or global
+ * @error: (out): place to store error (if any)
+ *
+ * Returns: whether progress reporting was successfully initialized or not
+ */
+gboolean bd_utils_init_prog_reporting_thread (BDUtilsProgFunc new_prog_func, GError **error __attribute__((unused))) {
+    /* XXX: the error attribute will likely be used in the future when this
+       function gets more complicated */
+
+    thread_prog_func = new_prog_func;
+
+    return TRUE;
+}
+
+static void thread_progress_deinitialized (guint64 task_id __attribute__((unused)), BDUtilsProgStatus status __attribute__((unused)), guint8 completion __attribute__((unused)), gchar *msg __attribute__((unused))) {
```

Comment:
I think ``thread_progress_muted()`` was a better name for this function.

Reply:
Ok, changed. It was just because it's used in the test of `bd_utils_prog_reporting_initialized`.